Detectors/MUON/MID/Clustering/test/bench_Clusterizer.cxx

```diff
@@ -20,81 +20,112 @@
 #include "DataFormatsMID/StripPattern.h"
 #include "Clusterizer.h"
 
-std::vector<o2::mid::ColumnData> generateTestData(int deId, const o2::mid::Mapping& midMapping)
+o2::mid::ColumnData& getColumn(std::vector<o2::mid::ColumnData>& patterns, int icolumn, int deId)
```

Comment:
Ok.
Actually, the explicit cast is done for a specific reason.
ColumnData is a data format describing the patterns that we expect from the CRU. It will be therefore packed into messages and transmitted from one device to another. For this kind of format, I tend to use the minimum number of bytes needed.
Since the information can be coded on uint8_t, this is what I use.
On the other hand, MpStripIndex is a temporary object that is not meant to be transferred. In this case, I tend to use the simpler type available (int), since it does not have an impact on the data flow.
Some explicit cast might be needed in this case, but in principle an implicit cast should be performed automatically (and anyway the information will always fit a uint8_t, so there should be no possibility of truncating the value in the conversion).

Cheers,
Diego

Reply:
ok, however it would be good to allow the compiler to check on that. 

anyhow, you should not use C-casts, and the O2 coding conventions say so (but it's not enforced by the code checker).